book/src/development/hir_ids.md

```diff
@@ -0,0 +1,78 @@
+# `HirId` in Clippy
+
+`HirId`s are identifiers for [fine-grained entities][fine_grained_entities]
+in the [HIR Map][hir_map] of the high-level intermediate representation (HIR),
+which is what we work with during a [LateLintPass].
+
+`HirId`s are everywhere in `rustc`. If we open up the Clippy
+codebase in a text editor and search `hir_id`,
+we will get numerous search results across hundreds of files.
+
+For instance, variables in `rustc` are identified using the `HirId`.
+The variable name alone is not enough, since variables can be shadowed.
+
+> Note: Besides `HirId`, the HIR has other identifiers such as
+> `DefId`, `LocalDefId` and `BodyId`.
+> Read more about [identifiers in the HIR][identifiers_in_hir].
+
+## Working with `HirId`s in Clippy
+
+A curious and observant reader of Clippy book might have noticed
+that `hir_id`s are hiding right under our noses: the [Expr][hir_expr]
+in `LateLintPass`'s [check_expr][late_lint_pass_check_expr]
+method contains a `hir_id` field:
+
+```rust
+pub struct Expr<'hir> {
+    pub hir_id: HirId,
+    pub kind: ExprKind<'hir>,
+    pub span: Span,
+}
+```
+
+Alas, `HirId`s are very useful to Clippy developers. Below are some
+examples for when we might need `HirId`s.
+
+## `find_parent_node` method
```

Comment:
Wait, this chapter is not at all about `hir_id`s but rather about util functions using `hir_id`s. I'm really unsure if this chapter is necessary. We should have a list with abbreviations and terms, that lists the `HirId` with a lint to the `rustc` HIR identifier documentation. This should be enough.

A chapter with a list of useful functions might be good. But this can also be really bitrotty.